conan/tools/gnu/gnutoolchain.py

```diff
@@ -214,7 +213,16 @@ def _initialize_default_extra_env(self):
                                   "RANLIB": ":",
                                   "STRIP": ":"}
             extra_env_vars.update(self._resolve_compilers_mapping_variables())
-
+        # Issue related: https://github.com/conan-io/conan/issues/15486
+        if self._is_cross_building and self._conanfile.conf_build:
+            compilers_build_mapping = (
+                self._conanfile.conf_build.get("tools.build:compiler_executables", default={},
+                                               check_type=dict)
+            )
+            if "c" in compilers_build_mapping:
+                extra_env_vars["CC_FOR_BUILD"] = compilers_build_mapping["c"]
+            if "cpp" in compilers_build_mapping:
+                extra_env_vars["CXX_FOR_BUILD"] = compilers_build_mapping["cpp"]
```

Comment:
Moved from `_resolve_compilers_mapping_variables`, which was only being applied if not `extra_env_vars`. It should not depend on that variable at all.